Makefile

```diff
@@ -16,37 +16,48 @@ export DRIVERS_DIR POLYCHORD_DIR PYPOLYCHORD_DIR LIKELIHOOD_DIR EXAMPLES_DIR BIN
 
 PYTHON=python
 
-# Whether to use MPI
+# Whether to use MPI (default to MPI on Linux)
 ifeq "$(shell uname)" "Linux"
-MPI=1
+	MPI ?= 1
 else
-MPI= 
+	MPI ?= 0
 endif
 
-# Whether to compile in debugging mode
-DEBUG=
+# Whether to compile in debugging mode (default: false)
+DEBUG ?= 0
 export MPI DEBUG
 
 # We can autodetect the compiler type on unix systems via the shell.
 # if you want to override this then just run make with
 # make COMPILER_TYPE=<your type>
 # where <your type> is gnu or intel
 ifeq "$(shell which ifort >/dev/null 2>&1; echo $$?)" "0" 
-COMPILER_TYPE=intel
+	COMPILER_TYPE=intel
 else ifeq "$(shell which gfortran >/dev/null 2>&1; echo $$?)" "0"
-COMPILER_TYPE=gnu
+	COMPILER_TYPE=gnu
+endif
+
+# Reset default values because the get set later
```

Comment:
The idea is that if no compilers are specified it should default to reasonable choices. The problem is that under macOS the compiler variables CC, CXX, and FC are defined when running the makefile. The main problem here is FC, since f77 is pretty useless.

Reply:
What was wrong with how the defaults were set in the previous versions of the makefiles? i.e. by having `CC=gfortran` in `Makefile_gnu`, rather than the if statements that are in there at the moment?

Reply:
The setup in master doesn't allow to switch between different compiler versions (e.g., gcc 8 and gcc 9) since `CC=gfortran` overwrites whatever `FC` I set. Using `FC ?= gfortran` doesn't work either because `make` sets `FC` to `f77` if not explicitly specified by the user.

Reply:
This surprises me a little, since these kind of commands have always overwritten any defaults for me:

```bash
make CC=gcc-8
#or 
export CC=gcc-8
make
```
Is this something that you've found not to work? I thought that the previous setup was pretty standard for makefiles.

Reply:
I just tried this on master and only `make FC=gfortran-8` seems to work. Using environmental variables, neither `export FC=gfortran-8 && make` nor `FC=gfortran-8 make` work. `setup.py` uses environmental variables to pass the compilers around.

Reply:
Ah, I see the issue now. The trouble is that it's [bad practice](https://www.gnu.org/software/make/manual/html_node/Environment.html) to have makefiles read environment variables in this way, and that's what the `-e` option is for. I've pushed an update that I believe should solve both problems.